### ISSUE TITLE
[BUG] Tinymce font loading over HTTPS gives mixed content error.

### ISSUE THREAD

**Issue by sten** [thedevdojo/voyager#4137]
## Version information

 - Laravel: v5.8.10
 - Voyager: v1.2.3
 - PHP: 7.2.1
 - Database: Mysql 5.7.22

## Description
When voyager is used over HTTPS and you load the create or edit form of a BREAD with a rich text editor, the font of Tinymce is not loaded. This results in a broken UI, i.e. the buttons do not have any icons.

## Steps To Reproduce
1. Host Voyager over HTTPS
2. Go to the create or edit of a BREAD with a rich text field
3. Check the rich text editor for missing icons
4. Check the browser console for errors, like: 
- Mixed Content: The page at 'https://www.mydomain.com/admin/pages/create' was loaded over HTTPS, but requested an insecure font 'http://www.mydomain.com/admin/assets?path=js/skins/voyager/fonts/tinymce.woff'. This request has been blocked; the content must be served over HTTPS.
- Mixed Content: The page at 'https://www.mydomain.com/admin/pages/create' was loaded over HTTPS, but requested an insecure font 'http://www.mydomain.com/admin/assets?path=js/skins/voyager/fonts/tinymce.ttf'. This request has been blocked; the content must be served over HTTPS.

## Expected behavior
The TinyMCE font should be properly loaded and the icons should be displayed in the editor.

## Additional context
I have been debugging this a bit. It appears the fonts are loaded in `skin.min.css` via the url() function of CSS. However `skin.min.css` is loaded from an ajax call from `app.js` via the assets REST service. Within app.js the assets REST service is called over HTTPS, which is correct. I assume the dynamically loaded skin.min.css does not have any HTTPS context.

**Comment by estebangallego** [comment]
Same issue, please advice. 
![Screen Shot 2019-06-18 at 6 48 09 PM](https://user-images.githubusercontent.com/5769433/59731081-b0c38a80-91f9-11e9-8bc5-d31ebd5c86a8.png)

**Comment by apiaget** [comment]
Hello,

I also encountered that issue.

I looked at my chrome inspector and tried to spot the differences between the fonts that load correctly and the tinymce one. I noticed that there is a slash (`/`) in the failed request between `voyager-assets` and `?path=...`.
A failed request : `https://karac.ch/admin/voyager-assets/?path=js/skins/voyager/fonts/tinymce.woff`
A successfull request : `https://karac.ch/admin/voyager-assets?path=fonts/voyager.woff`

See the screenshots :

![tinymce-2](https://user-images.githubusercontent.com/634083/60806533-8ffcae00-a183-11e9-9ec1-816baf72b889.JPG)

![tinymce-3](https://user-images.githubusercontent.com/634083/60806536-925f0800-a183-11e9-8447-a8e2a6267256.JPG)


I then changed my `skin.min.css` file and removed that slash (`/`). The TinyMCE font loads correctly now :


![tinymce-4](https://user-images.githubusercontent.com/634083/60806669-e833b000-a183-11e9-8a4b-3bdd4bb741a1.JPG)

```
The actual skin.min.css
@font-face{
  font-family:'tinymce';
  src:url('voyager-assets/?path=js/skins/voyager/fonts/tinymce.eot');
  src:url('voyager-assets/?path=js/skins/voyager/fonts/tinymce.eot?#iefix') format('embedded-opentype'), url('voyager-assets/?path=js/skins/voyager/fonts/tinymce.woff') format('woff'), url('voyager-assets/?path=js/skins/voyager/fonts/tinymce.ttf') format('truetype'), url('voyager-assets/?path=js/skins/voyager/fonts/tinymce.svg#tinymce') format('svg');
  font-weight:normal;
  font-style:normal
}
@font-face{
  font-family:'tinymce-small';
  src:url('voyager-assets/?path=js/skins/voyager/fonts/tinymce-small.eot');
  src:url('voyager-assets/?path=js/skins/voyager/fonts/tinymce-small.eot?#iefix') format('embedded-opentype'), url('voyager-assets/?path=js/skins/voyager/fonts/tinymce-small.woff') format('woff'), url('voyager-assets/?path=js/skins/voyager/fonts/tinymce-small.ttf') format('truetype'), url('voyager-assets/?path=js/skins/voyager/fonts/tinymce-small.svg#tinymce') format('svg');
  font-weight:normal;
  font-style:normal
}
```


```
Here are the modifications I did :
@font-face{
font-family:'tinymce';
  src:url('voyager-assets?path=js/skins/voyager/fonts/tinymce.eot');
  src:url('voyager-assets?path=js/skins/voyager/fonts/tinymce.eot?#iefix') format('embedded-opentype'), url('voyager-assets?path=js/skins/voyager/fonts/tinymce.woff') format('woff'), url('voyager-assets?path=js/skins/voyager/fonts/tinymce.ttf') format('truetype'), url('voyager-assets?path=js/skins/voyager/fonts/tinymce.svg#tinymce') format('svg');
  font-weight:normal;
  font-style:normal
}
@font-face{
  font-family:'tinymce-small';
  src:url('voyager-assets?path=js/skins/voyager/fonts/tinymce-small.eot');
  src:url('voyager-assets?path=js/skins/voyager/fonts/tinymce-small.eot?#iefix') format('embedded- opentype'), url('voyager-assets?path=js/skins/voyager/fonts/tinymce-small.woff') format('woff'), url('voyager-assets?path=js/skins/voyager/fonts/tinymce-small.ttf') format('truetype'), url('voyager-assets?path=js/skins/voyager/fonts/tinymce-small.svg#tinymce') format('svg');
  font-weight:normal;
  font-style:normal
}
```

Is this the correct correction for this problem ?

**Comment by bluscience** [comment]
I had the same issue and after investigating it I found that the problem that was caused by an nginx condition:

```
 # removes trailing slashes (prevents SEO duplicate content issues)
if (!-d $request_filename) {
	rewrite ^/(.+)/$ /$1 permanent;
}
```

After changing the nginx statement to remove the tailing slashes everywhere, except the admin area, the fonts loaded correctly.

```
# removes trailing slashes (prevents SEO duplicate content issues)
if ($uri !~ /admin) {
     rewrite ^/(.+)/$ /$1 permanent;
}
```

**Comment by sten** [comment]
It is indeed caused by the trailing slash, @apiaget. I could not convince the sys admin to edit the nginx conditions, so we fixed it in our Laravel project by overriding CSS fonts and classes.

However this should be fixed in Voyager itself. 

Thanks, @apiaget and @bluscience  for the help!

**Comment by estebangallego** [comment]
Making those changes on the skin.min.css file has solved my issue!

Thank you @apiaget !

**Comment by emptynick** [comment]
Fixed in https://github.com/the-control-group/voyager/pull/4464 (to be released in 1.3)
